packages/react/src/components/Checkbox/index.tsx

```diff
@@ -36,23 +36,35 @@ const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
       'aria-describedby': ariaDescribedby,
       disabled = false,
       checked = false,
+      indeterminate = false,
       children,
       ...other
     }: CheckboxProps,
     ref
   ): JSX.Element => {
     const [isChecked, setIsChecked] = useState(checked);
     const [focused, setFocused] = useState(false);
-    const checkRef = useRef<HTMLInputElement>(null);
+    const checkRef = useRef<HTMLInputElement | null>(null);
 
     useEffect(() => {
       setIsChecked(checked);
     }, [checked]);
 
     const refProp = ref || checkboxRef;
-    if (typeof refProp === 'function') {
-      refProp(checkRef.current);
-    }
+
+    const refCallback = (elem: HTMLInputElement | null): void => {
```

Comment:
Is there a reason this ref callback function was added?

If you're trying to access the ref, `useEffect` may be what you want here.

Reply:
Thanks, removed